s2wav/analysis.py

```diff
@@ -51,48 +53,71 @@ def analysis_transform(flm: np.ndarray,
         f_scal (np.ndarray): Array of scaling pixel-space coefficients
             with shape :math:`[n_{\theta}, n_{\phi}]`.
     """
-    J = samples.j_max(L)
+    flm = ssht_to_s2wav(ssht.forward(f, L, Reality=reality), L)
+    J = samples.j_max(L, lam)
 
-    f_wav_lmn = np.zeros(shapes.flmn_wav(L, N, J_min, lam), dtype = np.complex128)
-    f_scal_lm = np.zeros(shapes.flm_scal(L), dtype = np.complex128)
+    f_wav_lmn = np.zeros(shapes.flmn_wav(L, N, J_min, lam), dtype=np.complex128)
+    f_scal_lm = np.zeros(shapes.flm_scal(L), dtype=np.complex128)
 
     wav_lm, scal_l = filters.filters_directional(L, N, J_min, lam, spin, spin0)
-
     for j in range(J_min, J + 1):
-        for n in range(-N + 1, N + 1, 2):
-            for el in range(max(abs(spin), abs(n)), L + 1):
+        for n in range(-N + 1, N, 2):
+            for el in range(max(abs(spin), abs(n)), L):
                 if el != 0:
-                    psi = 8 * np.pi * np.pi / (2 * el + 1) * np.conj(wav_lm[j, el, L - 1 + n])
-                    for m in range (-el, el + 1):
-                        f_wav_lmn[j, N - 1 + n, el, L - 1 + m] = flm[el, L - 1 + m] * psi
-
-    for el in range(abs(spin), L +1):
+                    psi = np.conj(wav_lm[j, el, L - 1 + n])
+                    psi *= 8 * np.pi**2 / (2 * el + 1)
```

Comment:
@CosmoMatt Are you splitting the psi calculation over 2 lines so it's more readable or is there another reason?

Reply:
Nope, just so it's readable. Feel free to switch back if you prefer

Reply:
Will keep it as is, think the way you've written it is nicer.